lib/ansible/modules/cloud/amazon/ses_rule.py

```diff
@@ -0,0 +1,235 @@
+module: ses_rule
+short_description: Manages SES inbound receipt rules
+description:
+    - Allows creation, deletion, and management of SES receipt rules
+version_added: 2.4
+author:
+  - "Ben Tomasik (@tomislacker)"
+requirements: [ "boto3","botocore" ]
+options:
+  name:
+    description:
+      - The name of the receipt rule
+    required: True
+  ruleset:
+    description:
+      - The name of the receipt rule set
+    required: True
+  state:
+    description:
+      - Whether to create or destroy the receipt rule
+    required: False
+    default: present
+    choices: ["absent", "present"]
+  after:
+    description:
+      - Insert new rule before another (This only applies during creation)
+    required: False
+  enabled:
+    description:
+      - Whether the rule should be enabled or not
+    required: False
+    default: True
+  tls_required:
+    description:
+      - Whether inbound emails should require TLS
+    required: False
+    default: False
+  recipients:
+    description:
+      - Recipient specification(s)
+    required: True
+  actions:
+    description:
+      - Rule actions
+    required: True
+  scan_enabled:
+    description:
+      - Whether inbound emails should get virus scanned
+    required: False
+    default: False
+extends_documentation_fragment: aws
+"""
+
+EXAMPLES = """
+
+# Pushes emails to S3 that are received for any address @mydomain.com as well
+# as any address @<any subdomain>.mydomain.com
+- name: Create catch-all rule
+  ses_rule:
+    name: main-rule
+    ruleset: default-rule-set
+    recipients:
+      - '.mydomain.com'
+      - 'mydomain.com'
+    actions:
+      - S3Action:
+          BucketName: my-bucket
+
+"""
+
+RETURN = """
+changed:
+  description: >
+      if a SES rule has been created or deleted. NOTE: If a rule exists, it's
+      always considered a change whether or not one occurred due to the lack
+      of implementation in change detection.
+  returned: always
+  type: bool
+  sample:
+    changed: true
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import HAS_BOTO3
+from ansible.module_utils.ec2 import ec2_argument_spec
+from ansible.module_utils.ec2 import get_aws_connection_info
+from ansible.module_utils.ec2 import boto3_conn
+
+
+def rule_set_exists(ses_client, name):
+    rule_sets = ses_client.list_receipt_rule_sets()['RuleSets']
+    return len([s for s in rule_sets if s['Name'].lower() == name.lower()]) > 0
+
+
+def rule_exists(ses_client, ruleset, name):
+    rules = ses_client.describe_receipt_rule_set(RuleSetName=ruleset)['Rules']
+    return len([r for r in rules if r['Name'].lower() == name.lower()]) > 0
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        ruleset=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        after=dict(type='str', required=False),
```

Comment:
`type='str'` and `required=False` are both defaults, and neither is required in the argument spec (this applies anywhere either might occur)